language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/List.hs

```diff
@@ -251,19 +250,33 @@ enumFromTo = runQuote $ do
 -- > foldList {integer} {integer} addInteger 0
 sum :: (TermLike term TyName Name uni, uni `Includes` Integer) => term ()
 sum = runQuote $ do
-    let int = mkTyBuiltin @Integer ()
-        add = builtin () (BuiltinName () AddInteger)
-    return
-        . mkIterApp () (mkIterInst () foldList [int, int])
-        $ [ add , mkConstant @Integer () 0]
```

Comment:
Why change all of that? Just use eta-expansion to turn a builtin into a `Term` and avoid all the trouble.

Reply:
I think I did that before implementing the eta-expansion code.  I'll probably just leave it since it works.